client/state/reader/analytics/actions.js

```diff
@@ -1,12 +1,16 @@
+import { buildReaderTracksEventProps } from 'calypso/reader/stats';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { getReaderFollowsCount } from 'calypso/state/reader/follows/selectors';
 
-export const recordReaderTracksEvent = ( name, properties ) => ( dispatch, getState ) => {
-	const followsCount = getReaderFollowsCount( getState() );
-	dispatch(
-		recordTracksEvent( name, {
-			subscription_count: followsCount,
-			...properties,
-		} )
-	);
-};
+export const recordReaderTracksEvent =
```

Comment:
ngl, i have a negative opinion about TS that i keep trying to get over. 🤣 

Everything in this reader space seems to be non-TS and propTypes so that doesn't help for building a better option by working with it more. That said, I **_might_** open up a followup simply for this opportunity and ping you for review. But I'm going to send this and address a couple other items first.

Reply:
🤣
I'll take a quick stab at the refactor so I can get off my armchair! 😉